libs/api/api_internal.h

```diff
@@ -3,21 +3,67 @@
 
 #pragma once
 
+#include "ebpf_api.h"
 #include "ebpf_platform.h"
 #include "ebpf_windows.h"
 #include "spec_type_descriptors.hpp"
 
+struct ebpf_object;
+
+typedef struct ebpf_program
```

Comment:
I used "ebpf_program" so that we can add "struct ebpf_program" forward declaration in ebpf_api.h.

Doing that ensures the struct names (ebpf_program, ebpf_map, ebpf_object) match with the libbpf ones.

Reply:
you can still do that with `struct _ebpf_program` as needed.